core/src/main/java/org/apache/calcite/rel/rel2sql/PivotRelToSqlUtil.java

```diff
@@ -52,72 +59,89 @@ public class PivotRelToSqlUtil {
-      SqlNodeList axesNodeList, SqlNodeList inColumnList) {
-    SqlPivot sqlPivot = new SqlPivot(pos, query, axesNodeList, aggList, inColumnList);
+      SqlNodeList axisNodeList, SqlNodeList inColumnList) {
+    SqlPivot sqlPivot = new SqlPivot(pos, query, aggList, axisNodeList, inColumnList);
     SqlNode sqlTableAlias = sqlPivot;
     if (pivotTableAlias.length() > 0) {
       sqlTableAlias =
           SqlStdOperatorTable.AS.createCall(pos, sqlPivot,
-          new SqlIdentifier(pivotTableAlias, pos));
+              new SqlIdentifier(pivotTableAlias, pos));
     }
     SqlNode select =
         new SqlSelect(SqlParserPos.ZERO, null, null, sqlTableAlias,
-        builder.select.getWhere(), null,
-        builder.select.getHaving(), null, builder.select.getOrderList(),
-        null, null, SqlNodeList.EMPTY);
+            builder.select.getWhere(), null,
+            builder.select.getHaving(), null, builder.select.getOrderList(),
+            null, null, SqlNodeList.EMPTY);
     return select;
   }
 
-  private SqlNodeList getInValueNodes(Aggregate e) {
+  private SqlNodeList getInValueNodes(List<SqlNode> selectList, List<SqlNode> aggregateColList,
+      SqlNodeList axisNode) {
     SqlNodeList inColumnList = new SqlNodeList(pos);
-    for (AggregateCall aggCall : e.getAggCallList()) {
-      String columnName1 = e.getRowType().getFieldList().get(aggCall.filterArg).getKey();
-      String[] inValues = columnName1.split("'");
-      String tableInValueAliases = inValues[2];
-      if (tableInValueAliases.contains("null")) {
-        tableInValueAliases = tableInValueAliases.replace("_null", "")
-            .replace("-null", "")
-            .replace("'", "");
-      }
-      String[] columnNameAndAlias = tableInValueAliases.split("-");
-      SqlNode inListColumnNode;
-      if (columnNameAndAlias.length == 1) {
-        inListColumnNode = SqlLiteral.createCharString(inValues[1], pos);
-      } else {
-        pivotTableAlias = columnNameAndAlias[1];
-        if (columnNameAndAlias.length == 2) {
-          inListColumnNode = SqlLiteral.createCharString(inValues[1], pos);
-        } else {
-          inListColumnNode =
-              SqlStdOperatorTable.AS.createCall(
-                  pos, SqlLiteral.createCharString(
-                  inValues[1], pos),
-              new SqlIdentifier(columnNameAndAlias[2], pos));
-        }
-      }
-      inColumnList.add(inListColumnNode);
+
+    if (aggregateColList.size() == 0) {
+      selectList.stream()
+          .filter(x -> !(x instanceof SqlIdentifier))
+          .map(node -> {
+            // Extract the specific node as per your expression
+            SqlNode secondOperand = ((SqlBasicCall)
+                ((SqlCase)
+                    ((SqlBasicCall)
+                        ((SqlBasicCall) node)
+                            .getOperandList().get(0))
+                        .operand(0))
+                    .getWhenOperands().get(0))
+                .operand(1);
```

Comment:
this could be moved to method named extractValueItem() or any name of your choice

Reply:
i guess this looks fine , it is easy to debug and if we need filter conditions we can directly add there.